src/org/javarosa/xpath/expr/Encoding.java

```diff
@@ -10,13 +10,52 @@
   HEX("hex") {
     @Override
     String encode(byte[] bytes) {
-      return DatatypeConverter.printHexBinary(bytes).toLowerCase();
+      StringBuilder sb = new StringBuilder(bytes.length * 2);
+      for (byte b : bytes) {
+        sb.append(HEX_TBL[(b >> 4) & 0xF]);
+        sb.append(HEX_TBL[(b & 0xF)]);
+      }
+      return sb.toString();
     }
   },
   BASE64("base64") {
     @Override
-    String encode(byte[] bytes) {
-      return DatatypeConverter.printBase64Binary(bytes);
+    String encode(byte[] sArr) {
+      // Copied from: https://github.com/brsanthu/migbase64/blob/master/src/main/java/com/migcomponents/migbase64/Base64.java
```

Comment:
What a shame we have to include all this bit twiddling. I don’t have a better option.

Reply:
Eh :/